### PR TITLE
[ios] Enable SKU tokens for all requests

### PR DESCRIPTION
**Pull request by friedbunny**
- Removes temporary conditional code added in #14467 (which was itself a follow-up to https://github.com/mapbox/mapbox-gl-native/pull/14421).
- Other small cleanups.
- Still waiting on text for a changelog entry, but that can come later.

/cc @julianrex @zugaldia

### CODE REVIEW

File: platform/darwin/src/MGLAccountManager_Private.h [hunk 1]
```diff
@@ -16,7 +16,6 @@ NS_ASSUME_NONNULL_BEGIN
 #if TARGET_OS_IPHONE || TARGET_OS_SIMULATOR
 /// The current global SKU.
 @property (class, atomic, readonly, nullable) NSString *skuToken;
```
Comment (friedbunny) [inline platform/darwin/src/MGLAccountManager_Private.h]: I missed removing this `nullable` — will do soon.

Comment (friedbunny) [reply platform/darwin/src/MGLAccountManager_Private.h]: → https://github.com/mapbox/mapbox-gl-native/pull/14578/commits/a7b2d9e0c6315ac65443e71fc806922dd536c5b5

**Comment by zugaldia** [comment]
Tagging as `DO NOT MERGE` - I'll follow up with an update next week.

**Comment by mb12** [comment]
@friedbunny What is the difference between 'SKU Token' and 'Access Token'? Would this require any  existing applications / web-apps already using 'Access Token'  to be changed?

https://github.com/mapbox/mapbox-gl-native/blob/43553c37e98225d7951d0abb694b080075c1b4c4/platform/darwin/src/MGLAccountManager.m#L83

**Comment by friedbunny** [comment]
Updated the original post with information on why this PR exists.

**Pull request merged**